src/components/Slider/SliderBase/SliderBase.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { forwardRef } from "react";
+import PropTypes from "prop-types";
+import "./SliderBase.scss";
+import { useSliderActions, useSliderSelection, useSliderUi } from "../SliderContext";
+import { bem, calcDimensions, getNearest, moveToPx } from "../SliderHelpers";
+import { useSliderRail } from "../SliderHooks";
+import SliderRail from "./SliderRail";
+import SliderTrack from "./SliderTrack";
+import SliderFilledTrack from "./SliderFilledTrack";
+import SliderThumb from "./SliderThumb";
+import {
+  isArrowDownEvent,
+  isArrowLeftEvent,
+  isArrowRightEvent,
+  isArrowUpEvent,
+  isPageDownEvent,
+  isPageUpEvent,
+  isEndEvent,
+  isHomeEvent
+} from "../../../utils/dom-event-utils";
+
+function getKey(index) {
+  return index;
+}
+
+const SliderBase = forwardRef(({ className }, ref) => {
+  const { color, disabled, dragging, size, shapeTestId } = useSliderUi();
+  const { definePageStep, min, max, ranged, step, value } = useSliderSelection();
+  const { changeValue, increaseValue, decreaseValue } = useSliderActions();
+  const { railCoords, railRef } = useSliderRail({ min, max, step, ref });
+  const { dimension, offset, positions } = calcDimensions({ max, min, ranged, value });
+
+  function handlePointerMove(e) {
+    if (dragging === null) {
+      return;
+    }
+    const offsetInPx = Math.round(e.clientX - railCoords.left);
+    const newValue = moveToPx({ offsetInPx, min, max, railCoords, step });
+    changeValue(newValue);
+  }
+
+  function handleRailClick(e) {
```

Comment:
as this specific function is being passed to a child component - lets wrap it with useCallback - or lets open this issue up for discussion in the group ?

Reply:
Done!